src/components/pages/Site/Site.js

```diff
@@ -126,10 +126,22 @@ const SiteForm = ({
   handleLongitudeChange,
 }) => {
   const handleLngLatSwap = () => {
-    handleLatitudeChange(formik.getFieldProps('longitude').value)
-    handleLongitudeChange(formik.getFieldProps('latitude').value)
+    const currentLatitude = formik.getFieldProps('latitude').value
+    const currentLongitude = formik.getFieldProps('longitude').value
+
+    handleLatitudeChange(currentLongitude)
+    handleLongitudeChange(currentLatitude)
+
+    formik.setFieldTouched('longitude', true)
+    formik.setFieldTouched('latitude', true)
   }
 
+  // hack to ensure formik values are updated before manually triggering validation
```

Comment:
instead of a comment, can we name the effect? Either `const _someExplaningName = useEffect...` or `useEffect(function someExplainingName() {}, []}`

Not a blocker, but if we can avoid a comment, we should IMO.

Reply:
will update to a named effect